server/src/com/cloud/network/router/NetworkHelperImpl.java

```diff
@@ -509,7 +511,9 @@ public DomainRouterVO deployRouter(final RouterDeploymentDefinition routerDeploy
 
             if (startRouter) {
                 try {
-                    router = startVirtualRouter(router, _accountMgr.getSystemUser(), _accountMgr.getSystemAccount(), routerDeploymentDefinition.getParams());
+                    final Account caller = CallContext.current().getCallingAccount();
```

Comment:
@anshul1886 could you please create a method for the new lines? As it is a duplicate code, a method keeps the code clean and helps in future works (especially if well documented).